ports/esp8266/network_wlan.c

```diff
@@ -170,6 +170,37 @@ static mp_obj_t esp_disconnect(mp_obj_t self_in) {
 }
 static MP_DEFINE_CONST_FUN_OBJ_1(esp_disconnect_obj, esp_disconnect);
 
+static mp_obj_t build_stations_tuple(void) {
+    mp_uint_t count = wifi_softap_get_station_num();
+    if (count == 0) {
+        return mp_const_empty_tuple;
+    }
+    mp_obj_tuple_t *stations = NULL;
+    struct station_info *info = wifi_softap_get_station_info();
+    if (info == NULL) {
+        mp_raise_msg(&mp_type_OSError, MP_ERROR_TEXT("no station info available"));
+    }
+    nlr_buf_t nlr;
+    if (nlr_push(&nlr) == 0) {
+        stations = MP_OBJ_TO_PTR(mp_obj_new_tuple(count, NULL));
+        struct station_info *current = info;
+        size_t station_index = 0;
+        mp_obj_t entry[2] = { 0 };
+        while (current != NULL && station_index < count) {
+            entry[0] = mp_obj_new_bytes(current->bssid, sizeof(current->bssid));
+            entry[1] = netutils_format_ipv4_addr((uint8_t *)&current->ip.addr, NETUTILS_BIG);
+            stations->items[station_index++] = mp_obj_new_tuple(2, entry);
+            current = STAILQ_NEXT(current, next);
+        }
+        nlr_pop();
+    } else {
+        mp_obj_print_exception(&mp_plat_print, MP_OBJ_FROM_PTR(nlr.ret_val));
+        stations = NULL;
+    }
+    wifi_softap_free_station_info();
+    return stations != NULL ? MP_OBJ_FROM_PTR(stations) : MP_OBJ_NULL;
```

Comment:
You can't return `MP_OBJ_NULL` from a method, that will crash the runtime.

Reply:
I remember doing that in the past for other code of mine and it worked, but I don't recall whether it worked by accident or not (I recall doing something with `None` return values, but it's been a while).

Anyway, using the NLR callback made this bit of code disappear :)